force-app/main/default/classes/TaskTriggerHandler.cls

```diff
@@ -0,0 +1,67 @@
+public without sharing class TaskTriggerHandler {
+  public static void updateRelatedAccounts(List<Task> tasks) {
+    Map<Id, Account> accountMap = countSpecificTasks(
+      getUnitedStatesProspectAccounts(tasks)
+    );
+    update accountMap.values();
+  }
+
+  public static List<Account> getUnitedStatesProspectAccounts(
+    List<Task> tasks
+  ) {
+    Set<Id> accIds = new Set<Id>();
+    for (Task task : tasks) {
+      accIds.add(task.whatId);
+    }
+
+    Id recordTypeId = Schema.SObjectType.Account.getRecordTypeInfosByDeveloperName()
+      .get('United_States')
+      .getRecordTypeId();
+
+    List<Account> accountsList = [
+      SELECT
+        Id,
+        Open_Tasks_Call__c,
+        Open_Tasks_Email__c,
+        Open_Tasks_Meeting__c,
+        Open_Tasks_Other__c,
+        (SELECT WhatId, Type FROM Tasks WHERE isClosed = FALSE)
+      FROM Account
+      WHERE Id IN :accIds AND RecordTypeId = :recordTypeId AND Type = 'Prospect'
+    ];
+
+    return accountsList;
+  }
+
+  public static Map<Id, Account> countSpecificTasks(List<Account> accounts) {
```

Comment:
Do you have a reason for Map?

Reply:
I thought that without Map my code will only work for one account, but with new implementation map is not needed.